File: entities/note.go
package entities

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type NoteVisibility string

const (
	NoteVisibilityInternal        NoteVisibility = "INTERNAL"
	NoteVisibilityVisibleToDoctor NoteVisibility = "VISIBLE_TO_DOCTOR"
)

type DoctorNote struct {
	ID         uuid.UUID      `gorm:"primaryKey" json:"id"`
	DoctorID   uuid.UUID      `gorm:"not null;index" json:"doctor_id"`
	AdminID    uuid.UUID      `gorm:"not null;index" json:"admin_id"`
	Note       string         `gorm:"type:text;not null" json:"note"`
	Visibility NoteVisibility `gorm:"size:50;default:'INTERNAL';not null" json:"visibility"`
	CreatedAt  time.Time      `gorm:"autoCreateTime" json:"created_at"`
}

func (n *DoctorNote) BeforeCreate(tx *gorm.DB) (err error) {
	if n.ID == uuid.Nil {
		n.ID = uuid.New()
	}
	return nil
}

File: python_backend/entities/note.py
import enum
import uuid
from sqlalchemy import Column, String, DateTime, Text, Enum, ForeignKey
from sqlalchemy.sql import func
from database import Base

class NoteVisibility(str, enum.Enum):
    INTERNAL = "INTERNAL"
    VISIBLE_TO_DOCTOR = "VISIBLE_TO_DOCTOR"

class DoctorNote(Base):
    __tablename__ = "doctor_notes"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    doctor_id = Column(String(36), ForeignKey("doctors.id", ondelete="CASCADE"), nullable=False, index=True)
    admin_id = Column(String(36), nullable=False, index=True)
    note = Column(Text, nullable=False)
    visibility = Column(Enum(NoteVisibility, name="note_visibility_enum", native_enum=False), default=NoteVisibility.INTERNAL, nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
